sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -295,14 +300,16 @@
   private static final int DEFAULT_NUM_RETRIES = 6;
 
   /** Returns a new {@link Read} transform for reading from Kinesis. */
-  public static Read read() {
-    return new AutoValue_KinesisIO_Read.Builder()
-        .setMaxNumRecords(Long.MAX_VALUE)
-        .setUpToDateThreshold(Duration.ZERO)
-        .setWatermarkPolicyFactory(WatermarkPolicyFactory.withArrivalTimePolicy())
-        .setRateLimitPolicyFactory(RateLimitPolicyFactory.withoutLimiter())
-        .setMaxCapacityPerShard(ShardReadersPool.DEFAULT_CAPACITY_PER_SHARD)
-        .build();
+  public static Read<KinesisRecord> read() {
```

Comment:
@lukecwik I suggested that Piotr make Read generic so that we can add `Read<byte[]> readData()` naturally, but now I'm wondering if this is a bad idea since it changes our Public API. Does this have a risk of breaking users?

It looks like at least the way the method is used in our tests (`p.apply(KinesisIO.read())`) is unaffected.

Reply:
Prob. this is OK as long as the change is backwards compatible for user pipelines.

Reply:
The worst thing I can imagine is that a user will get a raw type warning if he used it like `KinesisIO.Read read = KinesisIO.read();`

Reply:
I think it should be fine if it doesn't require a user code changes.